devicetypes/qubino/qubino-dimmer.src/qubino-dimmer.groovy

```diff
@@ -0,0 +1,634 @@
+
+	if(!childDevice) {
+		log.debug "handleChildEvent / creating a child device"
+		def childComponentLabel	= "Qubino Temperature Sensor"
+		def childDthName		= "Child Temperature Sensor"
+		def childDthNamespace 	= "qubino"
+
+		createChildDevice(childDthNamespace, childDthName, childDni, childComponentLabel)
+		childDevice = childDevices.find { it.deviceNetworkId == childDni }
+	}
+	log.debug "handleChildEvent / sending event: ${map} to child: ${childDevice}"
+	childDevice?.sendEvent(map)
+}
+
+def createChildDevice(childDthNamespace, childDthName, childDni, childComponentLabel) {
+	try {
+		log.debug "Creating a child device: ${childDthNamespace}, ${childDthName}, ${childDni}, ${childComponentLabel}"
+		def childDevice = addChildDevice(childDthNamespace, childDthName, childDni, device.hub.id,
+				[
+					completedSetup: true,
+					label: childComponentLabel,
+					isComponent: false
+				])
+		log.debug "createChildDevice: ${childDevice}"
+		childDevice
+	} catch(Exception e) {
+		log.debug "Exception: ${e}"
+	}
+}
+
+def on() {
+	encapCommands([
+			zwave.basicV1.basicSet(value: 0xFF),
+			zwave.switchMultilevelV3.switchMultilevelGet()
+	], 3000)
+}
+
+def off() {
+	encapCommands([
+			zwave.basicV1.basicSet(value: 0x00),
+			zwave.switchMultilevelV3.switchMultilevelGet()
+	], 3000)
+}
+
+def setLevel(value) {
+	log.debug "setLevel >> value: $value"
+	def valueaux = value as Integer
+	def level = Math.max(Math.min(valueaux, 99), 0)
+	encapCommands([
+			zwave.basicV1.basicSet(value: level),
+			zwave.switchMultilevelV3.switchMultilevelGet()
+	], 3000)
+}
+
+def setLevel(value, duration) {
+	log.debug "setLevel >> value: $value, duration: $duration"
+	def valueaux = value as Integer
+	def level = Math.max(Math.min(valueaux, 99), 0)
+	def dimmingDuration = duration < 128 ? duration : 128 + Math.round(duration / 60)
+	def getStatusDelay = duration < 128 ? (duration * 1000) + 2000 : (Math.round(duration / 60) * 60 * 1000) + 2000
+	encapCommands([
+			zwave.switchMultilevelV3.switchMultilevelSet(value: level, dimmingDuration: dimmingDuration),
+			zwave.switchMultilevelV3.switchMultilevelGet()
+	], getStatusDelay)
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	log.debug "refresh"
+	refreshChild()
+	encapCommands(getRefreshCommands())
+}
+
+def getRefreshCommands() {
+	def commands = []
+	commands << zwave.basicV1.basicGet()
+
+	if(isFlushDimmer() || isDINDimmer()) {
+		commands << zwave.meterV2.meterGet(scale: 0)
+		commands << zwave.meterV2.meterGet(scale: 2)
+	}
+	commands
+}
+
+private refreshChild() {
+	// refresh a child temperature sensor (if available)
+	if(childDevices){
+		def childDni = "${device.deviceNetworkId}:2"
+		def childDevice = childDevices.find { it.deviceNetworkId == childDni }
+
+		if (childDevice != null) {
+			sendHubCommand(zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1, scale: 0).format())
```

Comment:
you don't need to `format` this to pass it to `sendHubCommand`

Reply:
also shouldn't this get encapsulated?

Reply:
Yup, it should. It's fixed now.